WC3ChampionsStatisticService.UnitTests/Hubs/WebsiteBackendHubTests.cs

```diff
@@ -0,0 +1,567 @@
+    {
+        var authService = new Mock<IW3CAuthenticationService>();
+        var connections = new ConnectionMapping();
+        var contextAccessor = new Mock<IHttpContextAccessor>();
+        var friendRequestCache = new FakeFriendRequestCache();
+        var personalSettingsRepo = new Mock<IPersonalSettingsRepository>();
+
+        var friendList = new Friendlist("User#1234");
+        friendList.Friends.Add("Friend#5678");
+        var friendListCache = new TestFriendListCache();
+        friendListCache.Upsert(friendList);
+        var friendRepository = new TestFriendRepository();
+        IFriendCommandHandler friendCommandHandler = new TestFriendCommandHandler(friendRepository, friendListCache);
+
+        var tracingService = new TracingService(new System.Diagnostics.ActivitySource("test"), new Mock<IHttpContextAccessor>().Object);
+
+        var hub = new WebsiteBackendHub(
+            authService.Object,
+            connections,
+            contextAccessor.Object,
+            friendRequestCache,
+            personalSettingsRepo.Object,
+            friendCommandHandler,
+            tracingService
+        );
+
+        var testUser = new WebSocketUser { BattleTag = "User#1234", ConnectionId = "conn1" };
+        connections.Add("conn1", testUser);
+
+        var mockClients = new Mock<IHubCallerClients>();
+        var mockCaller = new Mock<ISingleClientProxy>();
+        mockClients.Setup(clients => clients.Caller).Returns(mockCaller.Object);
+        typeof(Hub).GetProperty("Clients").SetValue(hub, mockClients.Object);
+        typeof(Hub).GetProperty("Context").SetValue(hub, new HubCallerContextMock("conn1"));
+
+        await hub.RemoveFriend("Friend#5678");
+
+        Assert.That(friendList.Friends, Does.Not.Contain("Friend#5678"));
+        mockCaller.Verify(
+            c => c.SendCoreAsync(
+                It.Is<string>(s => s.Contains("FriendResponseData")),
+                It.IsAny<object[]>(),
+                default
+            ),
+            Times.Once
+        );
+    }
+
+    [Test]
+    public async Task LoadFriendListAndRequestsTraced_SendsFriendListAndRequests()
+    {
+        var authService = new Mock<IW3CAuthenticationService>();
+        var connections = new ConnectionMapping();
+        var contextAccessor = new Mock<IHttpContextAccessor>();
+        var friendRequestCache = new FakeFriendRequestCache();
+        var personalSettingsRepo = new Mock<IPersonalSettingsRepository>();
+
+        var friendList = new Friendlist("User#1234");
+        var friendListCache = new TestFriendListCache();
+        friendListCache.Upsert(friendList);
+        var friendRepository = new TestFriendRepository();
+        IFriendCommandHandler friendCommandHandler = new TestFriendCommandHandler(friendRepository, friendListCache);
+
+        var tracingService = new TracingService(new System.Diagnostics.ActivitySource("test"), new Mock<IHttpContextAccessor>().Object);
+
+        var hub = new WebsiteBackendHub(
```

Comment:
There is a lot of repeated boilerplate. Could this either be a parameterized test or part of a helper or the BeforeTest?

Reply:
Reduced boilerplate, `+124 −327`